### PR TITLE
Use solution folder (if available) during question import

### LINKED ISSUE #1245: Support alternate folder for question import
Test case generation from importing programming questions follows the same workflow and uses the same files as the ones displayed to students attempting the question.

This means the template files provided to the students must have the correct syntax so a test case report is generated. For Python, blank files are fine, but for compiled languages, it might be necessary to define dummy functions.

Removing this restriction is possible by having instructors provide an alternate folder named `solution`, which will override `submission` for test case generation. They can place full solutions there for easier maintenance of the questions, but it just needs to be syntax correct.

The solution directory must be removed when evaluating student submissions.

This change should not break existing packages. If there is no solution directory, retain the template directory.

### PR DESCRIPTION
**Pull request by fonglh**
Fixes #1245.

Allows syntax errors and infinite loops in templates.

### CODE REVIEW

File: lib/autoload/course/assessment/programming_package.rb [hunk 1]
```diff
@@ -117,6 +115,31 @@ def submission_files=(files)
...
+  # like generating a report from the solution folder.
+  def remove_solution_files
+    remove_folder_files(SOLUTION_PATH)
+  end
+
+  # Gets the contents of all solution files.
+  #
+  # @return [Hash<Pathname, String>] A hash mapping the file path to the file contents of each
+  #   file.
+  def solution_files
```
Comment (weiqingtoh) [inline lib/autoload/course/assessment/programming_package.rb]: `get_solution_files` to be consistent.

**Comment by weiqingtoh** [comment]
If you are interested, you can also try to do some meta-programming to declare `get_x_files` or `remove_x_files`.

**Comment by weiqingtoh** [comment]
Otherwise, LGTM!

File: lib/autoload/course/assessment/programming_package.rb (hunk 1 above)
Comment (fonglh) [inline lib/autoload/course/assessment/programming_package.rb]: This is to be consistent with the existing function `submission_files`. There's also `submission_files=` for assignment, so renaming that to `get_submission_files` wouldn't be nice too.

**Comment by fonglh** [comment]
@weiqingtoh Summary at `programming_package.rb` has been updated!

**Comment by allenwq** [comment]
LGTM

**Comment by allenwq** [comment]
LGTM

**Pull request merged**